### ISSUE TITLE
Fixpoints in SynFun position create static errors

### ISSUE THREAD

**Issue by Negabinary** [hazelgrove/hazel#1299]
e.g. `(fix f -> fun x -> x + 1)(3)` on https://[hazel](https://hazel.org/build/remove-dhexp/).org/build/remove-dhexp/ branch. Gives "Expected a Constructor".

![image](https://github.com/hazelgrove/hazel/assets/31668468/f9516fb2-796a-4e61-ad56-8e918ea29c9d)

**Comment by disconcision** [comment]
Reminder for myself that I have notes in mine and matt's slack message history regarding this. There are some non-trivial design considerations to address in fixing this properly

**Comment by disconcision** [comment]
@Negabinary re: this: talked about this with cyrus a bit. some of me questionable design decisions notwithstanding we can patch this ftm with a custom mode logic for Fix that turns SynFun into Fun. in retrospect this is pretty much what all the functions in Mode are doing already

**Comment by disconcision** [comment]
@Negabinary similarly

**Comment by Negabinary** [comment]
@disconcision No, this is still an issue I think, at least I can reproduce it in this example:

![Image](https://github.com/user-attachments/assets/2c7dc9ab-7319-49af-85dc-55a413e05747)

I tried typing the example on this issue into Hazel but that seems to cause hazel to freeze now...

**Comment by disconcision** [comment]
Seems fixed
